test/engine-luatest/gh_6961_fkey_same_space_test.lua

```diff
@@ -91,6 +91,9 @@ g.test_foreign_key_primary = function(cg)
             "Foreign key 'filesystem' integrity check failed: tuple is referenced",
             function() fs:delete{11} end
         )
+
+        local fkey = box.space._space:get(fs.id).format[3].foreign_key.fkey
+        t.assert_equals(fkey.space, 0)
```

Comment:
I would add something like this:
```
tarantool> opts = setmetatable({}, { __serialize = 'map' }) 
---
...

tarantool> format = {{name = 'i', type = 'integer'}, {name = 'a', type = 'integer', foreign_key = {a1 = {space = 0, field = 'i'}}}}
---
...

tarantool> a = {512, 1, 'a', 'memtx', 0, opts, format}
---
...

tarantool> box.space._space:insert(a) 
---
- [512, 1, 'a', 'memtx', 0, {}, [{'name': 'i', 'type': 'integer'}, {'type': 'integer',
      'name': 'a', 'foreign_key': {'a1': {'space': 0, 'field': 'i'}}}]]
...

tarantool> box.space._space:get(512)
---
- [512, 1, 'a', 'memtx', 0, {}, [{'name': 'i', 'type': 'integer'}, {'type': 'integer',
      'name': 'a', 'foreign_key': {'a1': {'space': 0, 'field': 'i'}}}]]
...
```
I mean this problem is mainly for cases where you need to insert directly into _space (for example from SQL) so it makes sense to check if this works correctly. And, maybe, it makes sense to add similar tests for tuple FK.

Also, since this is labelled as a bug, I think it's better to create a separate file for this. Or at least replace the `bug` label with a `feature` label.

Reply:
Done